docs/framework-todo.md

```diff
@@ -0,0 +1,98 @@
+
+The complication is how to handle copying tiles from document to document. This is a key feature of CLUE. And a user can grab just one tile and copy it. If they grab a table that is displaying data from a shared virtual tile, then there could be a couple of options:
+- make the new tile and merge the shared data into the new tile. This way the new tile is completely self contained.
+- copy the visible tile and copy the virtual tile into the new document. 
+
+Then it gets more complicate if the user now copies another tile from the source document which was also connected to the virtual tile. A few options:
+- copy the second tile in a separated way so now the two tiles are no longer connected in the new document. This seems less than ideal since it seems likely that a user would expect them to continue to be connected.
+- copy the second tile and look for the copy of the virtual tile in the new document, then reconnect the second tile copy. If the user has manipulated the first tile copy in the meantime, this could get confusing.
+
+An alternative is to expose these virtual tiles and connections more explicitly. So when copying a tile the user could see that is connected to another tile in some way and decide what they want to do when they copy it.
+
+Another aspect of this is that the data being shared might not just a table. It might be a rich state with objects and arrays.
+
+Another part of this is how to work efficiently with large datasets. A proposal for this before was to use something like GraphQL to watch a shared dataset. This would allow any tiles viewing this dataset to define a query of the dataset so they wouldn't have to be notified of every dataset change, just the changes that update their query. This has the added benefit of allowing the dataset to be a remote or local and in both cases the tile works with GraphQL.
+
+**TODO:** What is the UX in CLUE Currently?
+
+# Improved Drag and Drop
+
+The HTML5 based drag and drop doesn't support thumbnail images the way that CLUE needs. It also is hard to write 
+
+# Updated tool state model
+
+It seems possible to both simplify what tools need to do to save their state and at the same time support recording all document changes. This would make it possible to replay how a student change the document, support a document level undo mechanism, and also pass state around as a list of changes like is currently done in several tools.
+
+The key change is to leverage MST to record these changes. This way a tool can just modify a simple state representation using MST actions. In the background MST records these changes at the document level. There are examples of using MST for undo redo which are built using JSON diffs and also include support for grouping changes together so an undo makes sense.
```

Comment:
In my thinking this goes in the general category of "leverage the capabilities of MST". I did look into this briefly at the very beginning but there was something about it (that I can no longer recall the details of) that made me think it would be more complicated or more time consuming to get started and so I went with the simple model of letting tiles define their own action models, but letting MST do more of the heavy lifting is appealing.